[SDTEST-3801] Add Test Parallelization documentation (#36985)
DataDog/documentation: content/en/tests/test_parallelization/best_practices.md

---
title: Test Parallelization Best Practices
further_reading:
  - link: "/tests/test_parallelization/setup/"
    tag: "Documentation"
    text: "Set up Test Parallelization"
  - link: "/tests/test_parallelization/configuration/"
    tag: "Documentation"
    text: "Configure Test Parallelization"
  - link: "/tests/test_parallelization/troubleshooting/"
    tag: "Documentation"
    text: "Troubleshooting Test Parallelization"
---

{{< callout url="https://www.datadoghq.com/product-preview/test-parallelization/" btn_hidden="false" header="Join the Preview!" >}}
Test Parallelization is in Preview. Complete the form to request access.
{{< /callout >}}

## Optimize the planning step

Test Parallelization adds a planning step that discovers tests before execution. For example, RSpec projects use dry-run discovery. Keep this step lightweight so the time saved by parallel execution is not offset by planning overhead.

### Preinstall system dependencies with Docker

> If your tests need operating system packages, bake them into a CI base image instead of installing them during every CI run.

Reviewer: ```suggestion
If your tests need operating system packages, include them in a CI base image instead of installing them during every CI run.
```
